pkg/templates/ignition.go

```diff
@@ -61,6 +63,26 @@ func (i *ignition) GenerateNode(kluster *kubernikusv1.Kluster, secret *v1.Secret
 		return nil, err
 	}
 
+	//this is the old default for backwards comptibility with clusters that don't have a passwort generated
+	//TODO: Remove once all klusters are upgraded
+	passwordHash := "$6$rounds=1000000$aldshc,xbneroyw$I756LN/FtceE1deC2H.tGeSdeeelaaZWRwzmbEuO1SANf7ssyPjnbQjlW/FcMvWGUGrhF64tX9fK0abE/4oQ80"
+	if nodePassword, ok := secret.Data["node-password"]; ok {
+		passwordCrypter := sha512_crypt.New()
+		//generate 16 byte random salt
+		salt, err := goutils.Random(16, 32, 127, true, true)
+		if err != nil {
+			return nil, fmt.Errorf("Unable to generate random salt: %s", err)
+		}
+		//We crank up the heat to 1 million rounds of hashing for this password (default 5000)
+		//Reason for this is we expose the resulting hash in the metadata service which is not very secure.
+		//It takes about 500ms on my workstation to compute this hash. So this means login to a node is also
+		// delayed for about a second which should be ok as this password is only meant as a last resort.
+		passwordHash, err = passwordCrypter.Generate(nodePassword, append([]byte("$6$rounds=1000000$"), salt...))
```

Comment:
How long and complicated is the resulting password? It often needs to be entered in the crappy VPC console without copy-paste...

Reply:
Its 12 chars and only characters and numbers. Its ok, I tested it.